### PR TITLE
zones are always retrieved from the existing shoot if it exists

### PR DESCRIPTION
**Pull request by Disper**
**Description**

Changes proposed in this pull request:

- zones are always retrieved from the existing shoot if it exists
- ~**DO NOT MERGE!** faces an issue with cluster creation (object has been modified)~ <- can't reproduce that anymore.
- ...

**Related issue(s)**

### CODE REVIEW

File: pkg/gardener/shoot/extender/dns.go [hunk 1]
```diff
@@ -107,6 +108,16 @@ func NewDNSExtender(secretName, domainPrefix, dnsProviderType string) func(runti
 
 		shoot.Spec.Extensions = append(shoot.Spec.Extensions, dnsExtension)
 
+		secretReference := gardener.NamedResourceReference{
```
Comment (akgalwas) [inline pkg/gardener/shoot/extender/dns.go:111]: I checked the validating webhook's [code](https://github.com/gardener/gardener-extension-shoot-dns-service/blob/master/pkg/admission/validator/shoot_validator.go#L57) once more. Now I'm sure we do the correct thing. Once the extension contains the secret reference the spec.resources field needs to be set accordingly.

File: pkg/gardener/shoot/converter.go [hunk 2]
```diff
@@ -17,12 +17,12 @@ type Converter struct {
 	config    config.ConverterConfig
 }
 
-func NewConverter(config config.ConverterConfig) Converter {
+func NewConverter(config config.ConverterConfig, shootFromState *gardener.Shoot) Converter {
```
Comment (akgalwas) [inline pkg/gardener/shoot/converter.go:20]: I'm not convinced to directly pass shoot to the converter. I would introduce some kind of "dynamic configuration". We need to align it with [Audit log PR](https://github.com/kyma-project/infrastructure-manager/pull/472) to have one concistent solution and make rebasing easier.

**Review by akgalwas — CHANGES_REQUESTED** [review]
In general the solution is OK, however, I believe we need to align the PR with ongoing Audit Log work. The way of enhancing converter with the context information should be discussed.

**Comment by Disper** [comment]
Closing in favor of #476 and  https://github.com/m00g3n/cluster-inventory/pull/6

**Pull request closed without merging**